RELEASES.md

```diff
@@ -2,10 +2,44 @@
 
-* * IMPORTANT: `eth_getProof` calls for historical state will be rejected by default.
-  * On archive nodes (`"pruning-enabled": false`): queries for historical proofs for state older than approximately 24 hours preceding the last accepted block will be rejected by default. This can be adjusted with the new option `historical-proof-query-window` which defines the number of blocks before the last accepted block which should be accepted for state proof queries, or set to `0` to accept any block number state query (previous behavior).
-  * On `pruning` nodes: queries for proofs past the tip buffer (32 blocks) will be rejected. This is in support of moving to a path based storage scheme, which does not support historical state proofs.
-* Added `Fortuna` upgrade as optional and defaulting to a `nil` timestamp.
+- Updated dockerhub image name to `avaplatform/avalanchego_subnet-evm` and tags to accommodate the new versioning scheme: {avalanchego version}_{subnet-evm version}
+
+## [v0.7.3](https://github.com/ava-labs/subnet-evm/releases/tag/v0.7.3)
+
+This version is backwards compatible to v0.7.0. It is optional, but encouraged.
+
+### AvalancheGo Compatibility
+
+The plugin version is unchanged at 39 and is compatible with AvalancheGo version v1.13.0.
+
+### Breaking changes
+
+- `eth_getProof` calls for historical state will be rejected by default (#1494: 4e5b9d86c617f046fd7ce36b9cbd76a6e27b4bc5)
+  - On archive nodes (`"pruning-enabled": false`): queries for historical proofs for state older than approximately 24 hours preceding the last accepted block will be rejected by default. This can be adjusted with the new option `historical-proof-query-window` which defines the number of blocks before the last accepted block which should be accepted for state proof queries, or set to `0` to accept any block number state query (previous behavior).
+  - On `pruning` nodes: queries for proofs past the tip buffer (32 blocks) will be rejected. This is in support of moving to a path based storage scheme, which does not support historical state proofs.
+
+### Features
+
+- Add `Fortuna` upgrade as optional and defaulting to a `nil` timestamp  (#1494, #1498)
+- Docker images Debian base upgraded from Bullseye (11) to Bookworm (12) (#1497)
+- AvalancheGo version upgraded from `v1.12.3-0.20250218154446-f1ec9a13b90a` to `v1.13.0`
+- plugin/evm/client: add `NewClientWithURL` function (#1509)
+
+### Fixes
+
+- Fix nil block possible bug in debug APIs (#1494: 0af9717087e6adafccf3486c770ae08cba09b751)
+
+### Documentation
+
+- plugin/evm: add local API docs (#1513)
+- 2023 Least Authority Audit (#1499)
+- precompile/contracts/warp:
+  - changes for 9000 upgrade (#1494)
+  - differentiate between L1 and Subnet self-signing optimization cases (#1529)
+- simulator: fix wrong flag name `staking-enabled` -> `--sybil-protection-enabled=false` (#1476)
+- typo fixes (#1491, #1506)
+
+**Full Changelog**: <https://github.com/ava-labs/subnet-evm/compare/v0.7.2...v0.7.3>
```

Comment:
A bit confused why these notes are in this PR..should they have been included when we released v0.7.3 a few weeks ago?

Reply:
they have been included, we just did not update them here. Maybe we should draft the release notes in this file before we actually cut and release the version. But the intention if this file is actually to keep pending changes for the next release, not to keep previous notes.

Reply:
updated it

Reply:
> we have seen that some context (like a critical change to API/flags) was missing in the release notes

That's why we could have a draft release notes where everyone can update it before a release. A bit more decoupled from the pull request though.

> because it was not possibly find the relevant context during the release process

IMO that's a commit message/description issue then

> I don't think a draft release would be ideal to be used as a documenting place

That's perhaps subjective, from my point of view, I would say the RELEASES.md file would be more overlooked (like in that case here) than the github releases 😄

Reply:
The problem here is not that we have forgotten to update the `RELEASES.md` in this PR, but we did not need to update it after the release and before this PR so the previous release was left under "Draft Release". I do think this `RELEASES.md` saves a ton of trouble in the release process and the one doing the release. Having a file make the changes reviewable/enforcable. We just need to improve our release process so we update this file + the actual version in `version.go` after cutting a new release.